src/o365/spo/commands/folder/folder-move.ts

```diff
@@ -0,0 +1,340 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+import * as url from 'url';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  sourceUrl: string;
+  targetUrl: string;
+  ignoreVersionHistory: boolean;
+  allowSchemaMismatch: boolean;
+}
+
+interface JobProgressOptions {
+  webUrl: string;
+  accessToken: string;
+  /**
+   * Response object retrieved from /_api/site/CreateCopyJobs
+   */
+  copyJopInfo: any;
+  /**
+   * Poll interval to call /_api/site/GetCopyJobProgress
+   */
+  progressPollInterval: number;
+  /**
+   * Max poll intervals to call /_api/site/GetCopyJobProgress
+   * after which to give up
+   */
+  progressMaxPollAttempts: number;
+  /**
+   * Retry attempts before give up.
+   * Give up if /_api/site/GetCopyJobProgress returns 
+   * X reject promises in a row
+   */
+  progressRetryAttempts: number;
+}
+
+class SpoFolderMoveCommand extends SpoCommand {
+  public get name(): string {
+    return commands.FOLDER_MOVE;
+  }
+
+  public get description(): string {
+    return 'Moves a folder to another location';
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: () => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let siteAccessToken: string = '';
+    const webUrl: string = args.options.webUrl;
+    const tenantUrl: string = `${url.parse(webUrl).protocol}//${url.parse(webUrl).hostname}`;
```

Comment:
It would be more efficient to parse the URL only once

Reply:
```suggestion
    const parsedUrl: url.UrlWithStringQuery = url.parse(webUrl);
    const tenantUrl: string = `${parsedUrl.protocol}//${parsedUrl.hostname}`;
```